File: frontend-admin-pro/src/api/equipment.ts
import { http } from '@/utils/http'
import type { ListResp } from '@/types/api'

export type EquipmentOut = {
  id: number
  code: string
  name: string
  model: string | null
  workshop: string | null
  status: string
  last_maintenance_date: string | null
  next_maintenance_date: string | null
  created_at: string
}

export type EquipmentMaintenancePlanOut = {
  id: number
  equipment_id: number
  plan_type: string
  check_items: string | null
  interval_days: number | null
  responsible_user_id: number | null
  next_date: string | null
  remark: string | null
  created_at: string
  updated_at: string
}

export type EquipmentMaintenanceLogOut = {
  id: number
  plan_id: number | null
  equipment_id: number
  check_result: string
  description: string | null
  attachments: string | null
  checked_by: number | null
  created_at: string
}

export const equipmentApi = {
  list(params?: { status?: string }) {
    return http.request<ListResp<EquipmentOut>>({ url: '/admin/equipment', method: 'GET', params })
  },
  create(data: { code?: string; name: string; model?: string; workshop?: string; remark?: string }) {
    return http.request<EquipmentOut>({ url: '/admin/equipment', method: 'POST', data })
  },
  update(
    equipmentId: number,
    data: {
      code?: string
      name?: string
      model?: string
      workshop?: string
      status?: string
      last_maintenance_date?: string | null
      next_maintenance_date?: string | null
      maintenance_interval_days?: number | null
      remark?: string
    },
  ) {
    return http.request<EquipmentOut>({ url: `/admin/equipment/${equipmentId}`, method: 'PUT', data })
  },
  check(equipmentId: number, params?: { check_type?: string; result?: string; description?: string }) {
    return http.request<void>({ url: `/admin/equipment/${equipmentId}/check`, method: 'POST', params })
  },
  listMaintenancePlans(params?: { equipment_id?: number }) {
    return http.request<ListResp<EquipmentMaintenancePlanOut>>({
      url: '/admin/equipment/maintenance-plans',
      method: 'GET',
      params,
    })
  },
  createMaintenancePlan(data: {
    equipment_id: number
    plan_type: string
    check_items?: string
    interval_days?: number
    responsible_user_id?: number
    next_date?: string
    remark?: string
  }) {
    return http.request<EquipmentMaintenancePlanOut>({
      url: '/admin/equipment/maintenance-plans',
      method: 'POST',
      data,
    })
  },
  updateMaintenancePlan(
    planId: number,
    data: {
      equipment_id?: number
      plan_type?: string
      check_items?: string
      interval_days?: number
      responsible_user_id?: number | null
      next_date?: string | null
      remark?: string
    },
  ) {
    return http.request<void>({ url: `/admin/equipment/maintenance-plans/${planId}`, method: 'PUT', data })
  },
  deleteMaintenancePlan(planId: number) {
    return http.request<void>({ url: `/admin/equipment/maintenance-plans/${planId}`, method: 'DELETE' })
  },
  listMaintenanceLogs(params?: { equipment_id?: number }) {
    return http.request<ListResp<EquipmentMaintenanceLogOut>>({
      url: '/admin/equipment/maintenance-logs',
      method: 'GET',
      params,
    })
  },
  createMaintenanceLog(data: {
    equipment_id: number
    plan_id?: number
    check_result: string
    description?: string
  }) {
    return http.request<EquipmentMaintenanceLogOut>({
      url: '/admin/equipment/maintenance-logs',
      method: 'POST',
      data,
    })
  },
}

File: backend/app/schemas/equipment.py
from datetime import date, datetime

from pydantic import BaseModel, Field


# ---------- 设备 ----------
class EquipmentCreateIn(BaseModel):
    code: str | None = Field(default=None, max_length=32)
    name: str = Field(min_length=1, max_length=128)
    model: str | None = Field(default=None, max_length=64)
    workshop: str | None = Field(default=None, max_length=64)
    remark: str | None = None


class EquipmentUpdateIn(BaseModel):
    code: str | None = Field(default=None, min_length=1, max_length=32)
    name: str | None = Field(default=None, min_length=1, max_length=128)
    model: str | None = Field(default=None, max_length=64)
    workshop: str | None = Field(default=None, max_length=64)
    status: str | None = Field(default=None, max_length=32)
    purchase_date: date | None = None
    last_maintenance_date: date | None = None
    next_maintenance_date: date | None = None
    maintenance_interval_days: int | None = None
    remark: str | None = None


class EquipmentOut(BaseModel):
    id: int
    tenant_id: int
    code: str
    name: str
    model: str | None
    workshop: str | None
    status: str
    purchase_date: date | None
    last_maintenance_date: date | None
    next_maintenance_date: date | None
    maintenance_interval_days: int | None
    remark: str | None
    created_at: datetime
    updated_at: datetime
    model_config = {"from_attributes": True}


# ---------- 设备保养计划 ----------
class EquipmentMaintenancePlanCreateIn(BaseModel):
    equipment_id: int
    plan_type: str = Field(min_length=1, max_length=16)  # daily/weekly/monthly
    check_items: str | None = None  # 检查项 JSON
    interval_days: int | None = None
    responsible_user_id: int | None = None
    next_date: date | None = None
    remark: str | None = None


class EquipmentMaintenancePlanUpdateIn(BaseModel):
    equipment_id: int | None = None
    plan_type: str | None = Field(default=None, min_length=1, max_length=16)
    check_items: str | None = None
    interval_days: int | None = None
    responsible_user_id: int | None = None
    next_date: date | None = None
    remark: str | None = None


class EquipmentMaintenancePlanOut(BaseModel):
    id: int
    tenant_id: int
    equipment_id: int
    plan_type: str
    check_items: str | None
    interval_days: int | None
    responsible_user_id: int | None
    next_date: date | None
    remark: str | None
    created_at: datetime
    updated_at: datetime
    model_config = {"from_attributes": True}


# ---------- 设备保养日志 ----------
class EquipmentMaintenanceLogCreateIn(BaseModel):
    plan_id: int | None = None
    equipment_id: int
    check_result: str = Field(min_length=1, max_length=32)
    description: str | None = None
    attachments: str | None = None  # JSON array


class EquipmentMaintenanceLogOut(BaseModel):
    id: int
    tenant_id: int
    plan_id: int | None
    equipment_id: int
    check_result: str
    description: str | None
    attachments: str | None
    checked_by: int | None
    created_at: datetime
    model_config = {"from_attributes": True}
